app/javascript/packages/document-capture/components/selfie-capture.jsx

```diff
@@ -33,6 +33,12 @@ function SelfieCapture({ value, onChange, errorMessage, className }, ref) {
   const { t } = useI18n();
   const labelRef = useRef(/** @type {HTMLDivElement?} */ (null));
   const wrapperRef = useRef(/** @type {HTMLDivElement?} */ (null));
+  const hadValue = useRef(false);
+  const isUpdated = useMemo(() => {
+    const nextIsUpdated = Boolean(value && hadValue.current);
+    hadValue.current = hadValue.current || Boolean(value);
+    return nextIsUpdated;
+  }, [value]);
```

Comment:
this is repeated across files, it's only 3 lines but it was a little dense for me to read, do you think it would be worth some sort of shared function/helper for this? Like some sort of:

```js
const hadValue = useRef(false);
const isUpdated = isUpdated({ value, hadValue }); // or whatever the calling syntax would be

/// other file
// helper to if value has been updated, needs `hadValue = useRef(false)` to be initialized outside
isUpdated = ({ value, hadValue }) => {
  return useMemo(() => {
    const nextIsUpdated = Boolean(value && hadValue.current);
    hadValue.current = hadValue.current || Boolean(value);
    return nextIsUpdated;
  }, [value]);
}
```

Reply:
> this is repeated across files, it's only 3 lines but it was a little dense for me to read, do you think it would be worth some sort of shared function/helper for this? Like some sort of:

The behavior is similar, but not quite the same. Selfie tracks if there had ever been a value previously, whereas file input tracks if the immediately value immediately prior had been assigned. This is because there's no way to take a new selfie without unsetting the existing selfie value.

I'd guess most of the confusion here might arise from the dance between `useRef` and `useMemo`, where abstracting the need to access the previous value (e.g. [`usePrevious`](https://usehooks.com/usePrevious/)) might improve the clarity a bit?

To a certain extent, I was willing to accept some "pain" of duplication here, using the discomfort as a motivator toward trying to reconcile this component as more of a specialized file input.

Reply:
Oh they are different, my bad 🙈 

And yeah, `usePrevious` seems like a good fit for these, but at the end of the day, I don't feel *that* strongly about cleaning these up, the names are fairly clear so it helps make it possible to understand

Reply:
`usePrevious` (74b3189) works pretty well for `FileInput` (8f37f76), and reads a fair bit better than it did IMO.

Couldn't find a good way to make it work with `SelfieCapture` which wouldn't be equally or more confusing than the current implementation, so going to leave that one as-is.